src/storage/slab/slab.c

```diff
@@ -189,23 +197,55 @@ _slab_table_update(struct slab *slab)
               heapinfo.nslab - 1);
 }
 
+static void
+_slab_init_lru(struct slab * slab, ptrdiff_t offset)
+{
+    TAILQ_FIRST(&heapinfo.slab_lruq) = (uint8_t *)slab + offset;
+    heapinfo.slab_lruq.tqh_last = NULL;
+}
+
+static struct slab *
+_slab_lruq_head(void)
+{
+    return TAILQ_FIRST(&heapinfo.slab_lruq);
+}
+
+static void
+_slab_update_lru(struct slab * slab, ptrdiff_t offset)
```

Comment:
I think this should be properly rebuilding the TAILQ using _INSERT macros.

Reply:
My intention here was to perform pointer arithmetic's for incorrect slab addresses in https://github.com/twitter/pelikan/blob/4e4fa90137a428c361c6039ffcb93513fe70c6a3/src/storage/slab/slab.h#L127 structures field.

TBH I am not sure how this could be achieved using INSERT macros.

Reply:
Solution updated - with extended TAILQ macros.